src/Microsoft.DotNet.Interactive.SqlServer.Tests/KqlFact.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Xunit;
+
+namespace Microsoft.DotNet.Interactive.SqlServer.Tests
+{
+    public sealed class KqlFact : FactAttribute
+    {
+        private const string TEST_KQL_CONNECTION_STRING = nameof(TEST_KQL_CONNECTION_STRING);
+        private static readonly string _skipReason;
+        
+        static KqlFact()
+        {
+            _skipReason = TestConnectionAndReturnSkipReason();
+        }
+        
+        public KqlFact()
+        {
+            if (_skipReason is not null)
+            {
+                Skip = _skipReason;
+            }
+        }
+        
+        private static string TestConnectionAndReturnSkipReason()
+        {
+            string clusterName = GetClusterForTests();
+            if (string.IsNullOrWhiteSpace(clusterName))
+            {
+                return $"Environment variable {TEST_KQL_CONNECTION_STRING} is not set. To run tests that require "
+                       + "SQL Server, this environment variable must be set to a valid connection string value.";
+            }
+
+            return null;
+        }
+        
+        public static string GetClusterForTests()
+        {
+            return "https://help.kusto.windows.net/"; //Environment.GetEnvironmentVariable(TEST_KQL_CONNECTION_STRING);
```

Comment:
If this is public, do we need to disable the tests in CI?

Reply:
this is now using the env variable

Reply:
If this works in CI without needing to install anything (as would be the case for SQL), do we need the environment variable? Can we just run these tests all the time?

Reply:
If you enable the tests they will prompt for authentication, so they would fail in CI, @justinMDotNet am I correct?

Reply:
Yeah there will be a pop up for authentication. I can spend some time today looking into a way to do that programmatically without any kind of user interaction.

Reply:
One solution for this would be for the Interactive app to be registered with Azure to get an App ID and then use that for authentication when running tests.